core/src/test/java/org/apache/brooklyn/core/catalog/internal/TestToSpecTransformer.java

```diff
@@ -20,43 +20,36 @@
 
 import java.util.Map;
 import java.util.Set;
-import java.util.concurrent.ConcurrentHashMap;
 
+import org.apache.brooklyn.api.catalog.BrooklynCatalog;
 import org.apache.brooklyn.api.catalog.CatalogItem;
 import org.apache.brooklyn.api.entity.Application;
 import org.apache.brooklyn.api.entity.EntitySpec;
 import org.apache.brooklyn.api.internal.AbstractBrooklynObjectSpec;
 import org.apache.brooklyn.api.mgmt.ManagementContext;
+import org.apache.brooklyn.api.typereg.RegisteredType;
 import org.apache.brooklyn.core.plan.PlanNotRecognizedException;
 import org.apache.brooklyn.core.plan.PlanToSpecTransformer;
-import org.apache.brooklyn.util.text.Identifiers;
+import org.apache.brooklyn.core.typereg.JavaClassNameTypePlanTransformer;
+import org.apache.brooklyn.core.typereg.RegisteredTypeLoadingContexts;
+import org.apache.brooklyn.util.collections.MutableSet;
 import org.apache.brooklyn.util.yaml.Yamls;
 
+import com.google.common.collect.Iterables;
+
 /**
- * Resolves previously registered specs by id.
- * First create a spec and register it, keeping the returned ID:
- * <pre> {@code
- * String specId = TestToSpecTransformer.registerSpec(EntitySpec.create(BasicEntity.class));
- * }</pre>
- *
- * Then build a plan to be resolved such as:
- * <pre> {@code
- *  brooklyn.catalog:
- *    id: test.inputs
- *    version: 0.0.1
- *    item: <specId>
- * } </pre>
+ * For use in conjunction with {@link StaticTypePlanTransformer};
+ * this will lookup an item by ID or in a map "type: id".
+ * <p>
+ * Should be removed when catalog is able to function using new-style lookups.
  */
 public class TestToSpecTransformer implements PlanToSpecTransformer {
```

Comment:
It's a good exercise to make this work.  Should be cleaner when we're completely migrated.